src/HttpClient/Plugin/ServerSideRequestForgeryProtection/ServerSideRequestForgeryProtectionPlugin.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Graby\HttpClient\Plugin\ServerSideRequestForgeryProtection;
+
+use Graby\HttpClient\Plugin\ServerSideRequestForgeryProtection\Exception\InvalidURLException;
+use Http\Client\Common\Plugin;
+use Http\Client\Exception\RequestException;
+use Http\Discovery\UriFactoryDiscovery;
+use Http\Message\UriFactory;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Validates each part of the URL against a white or black list, to help protect against Server-Side Request Forgery
+ * attacks.
+ *
+ * @see https://github.com/j0k3r/safecurl
+ * @see https://whitton.io/articles/safecurl-ssrf-protection-and-a-capture-the-bitcoins/
+ */
```

Comment:
Maybe, the plugin should be extract in his own repository

Reply:
Might a good idea.